google/cloud/pubsub_v1/subscriber/_protocol/streaming_pull_manager.py

```diff
@@ -174,18 +174,18 @@ def _process_requests(
         # Handle special errors returned for ack/modack RPCs via the ErrorInfo
         # sidecar metadata when exactly-once delivery is enabled.
         if errors_dict and ack_id in errors_dict:
-            exactly_once_error = errors_dict[ack_id]
+            exactly_once_error = str(errors_dict[ack_id])
```

Comment:
Instead of casting to a str here, we can change the annotation for the `errors_dict` parameter to Optional[Dict[str, str]]

See the map<str,str> here that we parse in `_get_ack_errors`: https://github.com/googleapis/googleapis/blob/34b98de13faa4813f2e818d7d3158912176a66b8/google/rpc/error_details.proto#L135

Also please change the return type of `_get_ack_errors` to Optional[Dict[str, str]] as well.